app/services/course_search_service.rb

```diff
@@ -51,44 +52,64 @@ def call
 
 private
 
-  def sites_with_distance_from_origin
+  def distance_with_university_area_adjustment
+    university_provider_type = Provider.provider_types[:university]
+    university_location_area_radius = 10
+    <<~EOSQL.gsub(/^[\s\t]*/, "").gsub(/[\s\t]*\n/, " ").strip
+      (CASE
+        WHEN provider.provider_type = '#{university_provider_type}'
+          THEN (distance - #{university_location_area_radius})
+        ELSE distance
+      END) as boosted_distance
+    EOSQL
+  end
+
+  def locatable_sites
     site_status = SiteStatus.arel_table
     sites = Site.arel_table
 
-    # Create virtual table with sites and site statuses
-    sites_with_status = site_status.join(sites).on(site_status[:site_id].eq(sites[:id]))
-
-    # only want new and running sites
-    new_and_running_sites = sites_with_status.where(site_status[:status].in(%w[new_status running]))
+    # Only running and published site statuses
+    running_and_published_criteria = site_status[:status].eq(SiteStatus.statuses[:running]).and(site_status[:publish].eq(SiteStatus.publishes[:published]))
 
     # we only want sites that have been geocoded
-    geocoded_new_and_running_sites = new_and_running_sites.where(sites[:latitude].not_eq(nil).and(sites[:longitude].not_eq(nil)))
+    has_been_geocoded_criteria = sites[:latitude].not_eq(nil).and(sites[:longitude].not_eq(nil))
 
     # only sites that have a locatable address
     # there are some sites with no address1 or postcode that cannot be
     # accurately geocoded. We don't want to return these as the closest site.
     # This should be removed once the data is fixed
-    locatable_new_and_running_sites = geocoded_new_and_running_sites.where(sites[:address1].not_eq("").or(sites[:postcode].not_eq("")))
+    locatable_address_criteria = sites[:address1].not_eq("").or(sites[:postcode].not_eq(""))
+
+    # Create virtual table with sites and site statuses
+    locatable_sites = site_status.join(sites).on(site_status[:site_id].eq(sites[:id]))
+
+    locatable_sites = locatable_sites.where(running_and_published_criteria)
+    locatable_sites = locatable_sites.where(has_been_geocoded_criteria)
+    locatable_sites.where(locatable_address_criteria)
+  end
 
+  def course_id_with_lowest_locatable_distance
     # select course_id and nearest site with shortest distance from origin
     # as courses may have multiple sites
     # this will remove duplicates by aggregating on course_id
-    courses_with_nearest_site = locatable_new_and_running_sites.project(:course_id, Arel.sql("MIN#{Site.distance_sql(OpenStruct.new(lat: origin[0], lng: origin[1]))} as distance")).group(:course_id)
+    locatable_sites.project(:course_id, Arel.sql("MIN#{Site.distance_sql(OpenStruct.new(lat: origin[0], lng: origin[1]))} as distance")).group(:course_id)
```

Comment:
Similar to some of the changes above, can we also break this down to make it a little more readable? Maybe with:

```ruby
    origin_lat_long = OpenStruct.new(lat: origin[0], lng: origin[1])
    site_distance_sql = Site.distance_sql(origin_lat_long)
```

Reply:
```suggestion
    origin_lat_long = OpenStruct.new(lat: origin[0], lng: origin[1])
    lowest_locatable_distance = Arel.sql("MIN#{Site.distance_sql(origin_lat_long)} as distance")
    locatable_sites.project(:course_id, lowest_locatable_distance).group(:course_id)
```

May that instead?